src/v/kafka/server/group.cc

```diff
@@ -3233,12 +3233,11 @@ void group::maybe_rearm_timer() {
 }
 
 ss::future<> group::do_abort_old_txes() {
-    if (!_catchup_lock->try_read_lock()) {
+    auto unit = _catchup_lock->attempt_read_lock();
```

Comment:
Oh it is done here, commit history needs fixing?

Reply:
imho commit history is alright, first commit switches group to use regular locks to fix the issue, second commit iterates on it and improves with `ssx:rwlocks`. Although I may switch them.